README.markdown

```diff
@@ -833,6 +833,20 @@ For example, if you want to run all jobs in the same process for cucumber, try:
 Resque.inline = ENV['RAILS_ENV'] == "cucumber"
 ```
 
+If you have an environment with a lot of tasks or special resource constraints
+around your Redis installation, you can configure Resque to share a redis
+connection between pre- and post-fork processes. This removes some of the
+safety provided by Resque's forking model, but can help moderate load and
```

Comment:
@isnotajoke could you expand more on:

> This removes some of the  safety provided by Resque's forking model

What risks are there? What things should the user look out for when using this option?

Also, do you know if this causes any issues with resque worker pools? I don't think detailing that out is a requirement for getting this merged, but if you happen to know, I think it would be great to add this in.

Reply:
I haven't observed any specific risks (or things to look out for). 

Decoupling task executions from one another (by forking, by sharing relatively little between parent & child processes across the fork) makes Resque very tolerant of even very destructive failures within individual tasks; the individual task execution may fail, but the worker process will remain functional. Adding more shared state across forks (as this feature does when enabled) chips away at that safety. Here, I'd be worried about tasks failing or exiting in a way that leaves the underlying Redis connection in a weird state, potentially causing issues for the worker process as it looks for work & updates queue state in redis, as well as for individual job executions. We never saw that in our local testing, but it is a risk (and something to consider if turning this on for a mature system with a lot of existing Resque tasks).

I think the explanation above is a lot for a README, but I'll try to add a little bit more detail on the risk to this change.